odl/solvers/functional/default_functionals.py

```diff
@@ -1684,11 +1684,11 @@ def __init__(self, *functionals):
         if (len(functionals) == 2 and
                 isinstance(functionals[1], Integral)):
             functionals = [functionals[0]] * functionals[1]
-        if any(functional.domain is ComplexNumbers()
+        if any(functional.range == ComplexNumbers()
```

Comment:
Would it make sense to give all sets an order-parameter? Something like Integers().hirachy_number = 0, RealNumbers().hirachy_number = 1, ComplexNumbers().hirachy_number = 2? Then the _highest priority_ of range would be easy without all those comparisons